lib/validator/v3/validator.go

```diff
@@ -202,10 +218,10 @@ func validateName(v *validator.Validate, topStruct reflect.Value, currentStructO
 	return nameRegex.MatchString(s)
 }
 
-func validateNamespacedName(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+func validatePortName(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
 	s := field.String()
-	log.Debugf("Validate namespacedName: %s", s)
-	return namespacedNameRegex.MatchString(s)
+	log.Debugf("Validate port name: %s", s)
+	return len(s) != 0 && len(k8svalidation.IsValidPortName(s)) == 0
```

Comment:
NO action required:
Not for this review, but we could report the errors that `IsValidPortName` returns ...

also weird that `IsValidPortName` doesn't check for length == 0 (but nothing to do about that since you're already handling it.

Reply:
The field validators are crap unfortunately and don't allow any information about the cause of the error to be propagated.  Could do a struct validation that would allow it, but I think that can wait.